packages/@react-spectrum/list/src/List.tsx

```diff
@@ -0,0 +1,139 @@
+  let {scale} = useProvider();
+  let collator = useCollator({usage: 'search', sensitivity: 'base'});
+  let layout = useMemo(() =>
+      new ListLayout<T>({
+        estimatedRowHeight: scale === 'large' ? 48 : 32,
+        estimatedHeadingHeight: scale === 'large' ? 33 : 26,
+        padding: 0, // TODO: get from DNA
+        collator
+      })
+    , [collator, scale]);
+
+  layout.collection = state.collection;
+  layout.disabledKeys = state.disabledKeys;
+  return layout;
+}
+
+function List<T extends object>(props, ref: DOMRef<HTMLDivElement>) {
+  let {
+    shouldUseVirtualFocus,
+    transitionDuration = 0
+  } = props;
+  let domRef = useDOMRef(ref);
+  let {collection} = useListState(props);
+
+  let gridCollection = new GridCollection({
+    columnCount: 1,
+    items: [...collection].map(item => ({
+      type: 'row',
+      childNodes: [{
+        ...item,
+        index: 0,
+        type: 'cell'
+      }]
+    }))
+  });
+
+
+  let state = useGridState({
+    ...props,
+    collection: gridCollection
+  });
+  let layout = useListLayout(state); // TODO should layout return nulls?
+
+  let collator = useCollator({usage: 'search', sensitivity: 'base'});
+  let {direction} = useLocale();
+  let keyboardDelegate = new GridKeyboardDelegate({
+    collection: state.collection,
+    disabledKeys: state.disabledKeys,
+    ref: domRef,
+    direction,
+    collator,
+    focusMode: 'cell'
+  });
+  let {gridProps} = useGrid({
+    keyboardDelegate,
+    ref: domRef
+  }, state);
+
+  let {styleProps} = useStyleProps(props);
+
+  // This overrides collection view's renderWrapper to support hierarchy of items in sections.
+  // The header is extracted from the children so it can receive ARIA labeling properties.
+  type View = ReusableView<Node<T>, unknown>;
+  let renderWrapper = (parent: View, reusableView: View) => (
+    <VirtualizerItem
+      key={reusableView.key}
+      reusableView={reusableView}
+      parent={parent} />
+  );
+
+  return (
+    <ListContext.Provider value={{state, keyboardDelegate}}>
+      <Virtualizer
+        {...gridProps}
+        {...styleProps}
+        ref={domRef}
+        focusedKey={state.selectionManager.focusedKey}
+        sizeToFit="height"
+        scrollDirection="vertical"
+        className={
+          classNames(
+            listStyles,
+            'react-spectrum-List',
+            styleProps.className
+          )
+        }
+        layout={layout}
+        collection={collection}
+        renderWrapper={renderWrapper}
+        transitionDuration={transitionDuration}
+        shouldUseVirtualFocus={shouldUseVirtualFocus}>
+        {(type, item) => {
+          if (type === 'item')  {
+            return (
+              <ListItem item={item} shouldUseVirtualFocus={shouldUseVirtualFocus} />
+            );
+          } else if (type === 'loader') {
+            return (<div>wait circle</div>);
+          } else if (type === 'placeholder') {
+            return (<div>placeholder</div>);
```

Comment:
Placeholders because we are waiting for designs? Or can we borrow/use the same stuff from ListBoxBase https://github.com/adobe/react-spectrum/blob/main/packages/@react-spectrum/listbox/src/ListBoxBase.tsx#L150-L162

Reply:
oh forgot about this, i can potentially borrow from there